core/src/credential/validation.rs

```diff
@@ -0,0 +1,32 @@
+use super::ContextValue;
+
+// Context must contain at least one URI
+// The first URI must be https://www.w3.org/2018/credentials/v1 (use BASE_CREDENDIAL_CONTEXT)
+pub fn credential_context_validation(
+    val: &Vec<ContextValue>,
+) -> Result<(), serde_valid::validation::Error> {
+    if val.is_empty() {
+        return Err(serde_valid::validation::Error::Custom(
+            "Context must contain at least one URI".to_string(),
+        ));
+    }
+
+    match val[0] {
```

Comment:
I'd probably craft this whole thing as 
```
match val[0] {
   error_condition => Err(...),
   error_condition2 => Err(...),
   _ => Ok(())
}
```

to avoid the explicit `return`s. You could make this whole method even  more concise by `match`-ing on `val.get(0)` and treating a `None` return value as the first "context must contain at least one URI" error intead of making that be a separate check

Reply:
Okay let me see about this

Reply:
This solution actually does make a lot of sense